onedal/datatypes/_data_conversion.py

```diff
@@ -51,18 +52,21 @@ def _convert_to_supported_impl(policy, *data):
         device = policy._queue.sycl_device
 
         def convert_or_pass(x):
-            if x.dtype is np.float64:
+            if x.dtype == np.float64:
+                warnings.warn("Data will be converted into float32 from "
+                              "float64 because device does not support it",
+                              RuntimeWarning, )
                 return x.astype(np.float32)
             else:
                 return x
 
         if not device.has_aspect_fp64:
-            return (convert_or_pass(x) for x in data)
+            return [convert_or_pass(x) for x in data]
         else:
-            return data
+            return [x for x in data]
 else:
     def _convert_to_supported_impl(policy, *data):
-        return data
+        return [x for x in data]
```

Comment:
Why list instead of tuple is returned now?

Reply:
Fixed